clients/oss-index/build.gradle.kts

```diff
@@ -30,5 +30,5 @@ dependencies {
     api(libs.retrofit)
 
     implementation(libs.bundles.kotlinxSerialization)
-    implementation(libs.retrofitConverterKotlinxSerialization)
+    implementation(libs.retrofit.converter.kotlinxSerialization)
```

Comment:
Should the new name be entirely Lower case always? 
e.g. in this case `libs.retrofit.converter.kotlinx.serialization`?

Reply:
I tried to reflect the project hierarchy of the dependencies to have a meaningful grouping and not replace every dash in the module name with a dot.
For example, in this case the group is "converter" which contains all the retrofit converters for Jackson, kotlinx.serialization and so on, but there are not multiple converters for different "kotlinx" libraries, so to me it does not make sense to introduce a group for that.
Similar for the comments below:
* "exposed-java" is not a group of modules, so I used "exposed-javaTime".
* "jiraRestClient" is the project, so I kept it together.
* I thought about "logj4-api", but since we don't have any other log4j dependencies I found it reasonable to use "log4Api" as the group.

Reply:
> but there are not multiple converters for different "kotlinx" libraries

True, but on the other hand it seems awkward to eliminate *camelCase* in some cases, but not others.

Reply:
> True, but on the other hand it seems awkward to eliminate camelCase in some cases, but not others.

To me the goal is not to eliminate camel-case but to group dependencies. Therefore I didn't apply any changes to dependencies which do not belong to any group.

In the `libs.versions.toml` it does indeed look a bit inconsistent to have dashes and camel-case in the names, but my focus was more on the `build.gradle.kts` files. For example, `retrofit-converter-kotlinxSerialization` might look not so nice, but the resulting `retrofit.convert.kotlinxSerialization` in the Kotlin code is totally fine to me. The approach is taken from a project created using this wizard: https://kmp.jetbrains.com/

Reply:
> In the `libs.versions.toml` it does indeed look a bit inconsistent to have dashes and camel-case in the names, but my focus was more on the `build.gradle.kts` files.

Ok, I could live with that, but I'd prefer if we could come up with some clear rules (that could also be implemented in code) on how to derive the TOML key name from the group / artifact name. But that seems to be difficult when looking at examples like

    exposed-javaTime = { module = "org.jetbrains.exposed:exposed-java-time", version.ref = "exposed" }

and

    jiraRestClient-api = { module = "com.atlassian.jira:jira-rest-java-client-api", version.ref = "jiraRestClient" }

where you need to understand the meaning of the dashed artifact name parts in order to apply some proper hierarchy.

Reply:
@fviernau: @sschuberth and me went through the change and agreed on the current state, if there is something you would like to change I think it would be better to do it in a separate smaller PR or to discuss it in the next core dev meeting.

Reply:
fine, sure just go ahead.